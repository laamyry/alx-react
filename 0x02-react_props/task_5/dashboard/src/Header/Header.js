import React from "react";
import logo from "../assets/holberton-logo.jpg";
import "./Header.css";

function Header() {
  return (
    <>
      <div className="Header">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1>School dashboard</h1>
        </div>
      </div>
    </>
  );
}

export default Header;
