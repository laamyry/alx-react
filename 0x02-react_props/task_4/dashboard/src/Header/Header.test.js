import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header component', () => {
  it('renders header correctly', () => {
    const wrapper = shallow(<Header />);

    // Check if the component renders the logo and h1 correctly
    expect(wrapper.find('.App-logo').exists()).toBeTruthy();
    expect(wrapper.find('h1').text()).toEqual('School dashboard');

    // Optional: Snapshot testing
    expect(wrapper).toMatchSnapshot();
  });
});
