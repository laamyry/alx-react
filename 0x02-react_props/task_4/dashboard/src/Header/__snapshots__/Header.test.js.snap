// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders header correctly 1`] = `ShallowWrapper {}`;
